src/riak_cs_s3_auth.erl

```diff
@@ -43,13 +44,21 @@
 
 -spec identify(RD::term(), #context{}) ->
                       {string() | undefined,
-                       string() | {v4, v4_attrs()} | undefined}.
+                       string() | {v4, v4_attrs()} | undefined} |
+                      {failed, Reason::atom()}.
 identify(RD,_Ctx) ->
     case wrq:get_req_header("authorization", RD) of
         undefined ->
             identify_by_query_string(RD);
         AuthHeader ->
-            parse_auth_header(AuthHeader)
+            %% Query string auth does not require date/x-amz-date header
```

Comment:
This branch is for Authorization header. Why does this comment mention about query string?